GPE-839 add check contact endpoint

nathanbegbie opened the pull request:
[Turn](https://whatsapp.praekelt.org/docs/index.html#contacts) gives us an endpoint to check that an msisdn is associated with a valid WhatsApp account.

This adds an endpoint that returns whether a number is valid or invalid.

NOTE: 
- this PR also refactors the View Tests within the Django `sidekick` app, so that each TestClass tests a single url/view
- it also refactors some of the sidekick/utils so that we can reuse the contact check call to Turn

nathanbegbie committed:
write failing tests for check contact view

TDD FTW

nathanbegbie committed:
Create view to check WhatsApp contact

- implement view that acts as a wrapper for https://whatsapp.praekelt.org/docs/index.html#templated-messages
- write docs about endpoint
- update changelog to include details

nathanbegbie committed:
write failing additional test for contact check endpoint

- handle an error response from Turn

nathanbegbie committed:
update check whatsapp contact view to ensure response type handled

could be a dict or string, in which case, pop it in a dict

erikh360 commented on sidekick/tests/test_views.py:
```diff
@@ -63,6 +45,44 @@ def test_health_endpoint_with_vars(self):
+
+
+class TestSendTemplateView(SidekickAPITestCase):
+    def add_whatsapp_messages_200_response(self, responses):
```
I'm not sure if we have to pass responses here?

nathanbegbie replied:
You're right :) thanks!

erikh360 commented on sidekick/views.py:
```diff
@@ -68,3 +70,40 @@ def send_wa_template_message(request):
+            "Content-Type": "application/json",
+        }
+
+        result = requests.post(
```
I'm not sure if you've seen this: https://github.com/praekeltfoundation/rp-sidekick/blob/develop/sidekick/utils.py#L27

We could apply some DRY here. 🙂

nathanbegbie replied:
Darn. I missed this. I'm going to have to refactor this a bit, but it will be better to it this way, thanks!

erikh360 reviewed (commented):
Nice, just the two comments.

nathanbegbie committed:
remove unnecessary responses reference in sidekick/test_views

nathanbegbie committed:
refactor get_whatsapp_contact function into get_whatsapp_contact_id

- rename get_whatsapp_contact to get_whatsapp_contact_id and find/replace
- create get_whatsapp_contact*s* function that acts as a wrapper for the Turn Contact Check API

nathanbegbie committed:
refactor CheckContactView to use get_whatsapp_contacts function

nathanbegbie commented:
@erikh360 ready for re-review

erikh360 reviewed (approved):
Nice!
